opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -463,51 +463,56 @@ EclipseGrid::EclipseGrid(const Deck& deck, const int * actnum)
             throw std::invalid_argument("file: " + fileName + " is not a valid egrid file, GRIDUNIT not found");
         }
 
-        const std::vector<std::string>& gridunit = egridfile.get<std::string>("GRIDUNIT");
-
-        const std::vector<int>& gridhead = egridfile.get<int>("GRIDHEAD");
-        const std::array<int, 3> dims = {gridhead[1], gridhead[2], gridhead[3]};
-
-        m_nx=dims[0];
-        m_ny=dims[1];
-        m_nz=dims[2];
+        {
+            const std::vector<int>& gridhead = egridfile.get<int>("GRIDHEAD");
 
-        const std::vector<float>& coord_f = egridfile.get<float>("COORD");
-        const std::vector<float>& zcorn_f = egridfile.get<float>("ZCORN");
+            this->m_nx = gridhead[1];
+            this->m_ny = gridhead[2];
+            this->m_nz = gridhead[3];
+        }
 
-        m_coord.assign(coord_f.begin(), coord_f.end());
-        m_zcorn.assign(zcorn_f.begin(), zcorn_f.end());
+        {
+            const std::vector<float>& coord_f = egridfile.get<float>("COORD");
+            const std::vector<float>& zcorn_f = egridfile.get<float>("ZCORN");
 
-        if (gridunit[0] != "METRES") {
+            m_coord.assign(coord_f.begin(), coord_f.end());
+            m_zcorn.assign(zcorn_f.begin(), zcorn_f.end());
+        }
 
+        if (const auto& gridunit = egridfile.get<std::string>("GRIDUNIT");
+            gridunit[0] != "METRES")
+        {
             constexpr auto length = ::Opm::UnitSystem::measure::length;
 
             if (gridunit[0] == "FEET"){
-                Opm::UnitSystem units(Opm::UnitSystem::UnitType::UNIT_TYPE_FIELD );
+                Opm::UnitSystem units(Opm::UnitSystem::UnitType::UNIT_TYPE_FIELD);
                 units.to_si(length, m_coord);
                 units.to_si(length, m_zcorn);
-            } else if (gridunit[0] == "CM"){
-                Opm::UnitSystem units(Opm::UnitSystem::UnitType::UNIT_TYPE_LAB );
+            }
+            else if (gridunit[0] == "CM"){
```

Comment:
space

Reply:
> space

Thanks for noticing.  I've pushed an update to address this.